### PR TITLE
Varnish Metrics Receiver

### PR DESCRIPTION
**Pull request by JonathanWamsley**
Add Support for Varnish Metrics.

[Scope Doc](https://docs.google.com/document/d/1wMdSs4DRGq1LOC0LNMgmtS-KbCXl1yCDTpV9YyQnFrQ/edit)

### CODE REVIEW

**Comment by sophieyfang** [comment]
Hi John PR#500 was merged in this morning.
The change is about unifying the data per integration into one single file metadata.yaml file.
See https://github.com/GoogleCloudPlatform/ops-agent/pull/513

File: docs/varnish.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Following the guide for [Configuring the Ops Agent](https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/configuration#file-location), add the required elements for your varnish instance configuration.
+
+To configure a receiver for your Varnish metrics, specify the following fields:
+
+| Field                 | Required | Default | Description                                                                                                                                                                 |
+|-----------------------|----------|---------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `type`                | required |         | Must be `varnish`.                                                                                                                                                          |
+| `collection_interval` | required |         | A [time.Duration](https://pkg.go.dev/time#ParseDuration) value, such as `30s` or `5m`.                                                                                      |
```
Comment (ridwanmsharif) [inline docs/varnish.md:17]: I was under the impression that this was optional and it defaulted to `60s`

See: https://github.com/GoogleCloudPlatform/ops-agent/blob/19e1eb67f06bf4497678df9a1f111cd1ca32f279/confgenerator/config.go#L393-L403

File: integration_test/third_party_apps_data/applications/varnish/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+short_name: varnish
+long_name: Varnish HTTP Cache
+expected_metrics:
```
Comment (ridwanmsharif) [inline integration_test/third_party_apps_data/applications/varnish/metadata.yaml:3]: Could we also add the expected logs to this file. See: https://github.com/GoogleCloudPlatform/ops-agent/pull/513

**Review by ridwanmsharif — COMMENTED** [review]
Overall looks good, left a couple of minor comments.

File: docs/varnish.md (hunk 1 above)
Comment (JonathanWamsley) [reply docs/varnish.md]: [couchdb](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/docs/couchdb.md?plain=1#L18) and [rabbitmq](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/docs/rabbitmq.md?plain=1#L22) appear to have the required field too for collection interval. Not sure if you want me to make that change in this pr too or if that would belong in another pr.

File: integration_test/third_party_apps_data/applications/varnish/metadata.yaml (hunk 2 above)
Comment (JonathanWamsley) [reply integration_test/third_party_apps_data/applications/varnish/metadata.yaml:3]: added `expected_logs:` field

**Review by ridwanmsharif — APPROVED** [review]

File: docs/varnish.md (hunk 1 above)
Comment (ridwanmsharif) [reply docs/varnish.md]: Separate PR is fine with me.

**Pull request merged**